travis fails on jsonQuerySubmitTest sometimes

I got this sometimes from travis, but not always:

edu.washington.escience.myria.systemtest.JsonQuerySubmitTest > jsonQuerySubmitTest FAILED
java.lang.AssertionError: expected:<SUCCESS> but was:<ACCEPTED>
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.failNotEquals(Assert.java:834)
at org.junit.Assert.assertEquals(Assert.java:118)
at org.junit.Assert.assertEquals(Assert.java:144)
at edu.washington.escience.myria.systemtest.JsonQuerySubmitTest.jsonQuerySubmitTest(JsonQuerySubmitTest.java:197)

--- a/src/edu/washington/escience/myria/parallel/QueryManager.java
+++ b/src/edu/washington/escience/myria/parallel/QueryManager.java
@@ -25,6 +25,7 @@ import edu.washington.escience.myria.MyriaConstants.FTMode;
 import edu.washington.escience.myria.RelationKey;
 import edu.washington.escience.myria.api.encoding.QueryEncoding;
 import edu.washington.escience.myria.api.encoding.QueryStatusEncoding;
+import edu.washington.escience.myria.api.encoding.QueryStatusEncoding.Status;
 import edu.washington.escience.myria.coordinator.catalog.CatalogException;
 import edu.washington.escience.myria.coordinator.catalog.MasterCatalog;
 import edu.washington.escience.myria.proto.ControlProto;
@@ -79,9 +80,10 @@ public class QueryManager {
   /**
    * @return if a query is running.
    * @param queryId queryID.
+   * @throws CatalogException if there is an exception checking the query status
    */
-  public boolean queryCompleted(final long queryId) {
-    return !runningQueries.containsKey(queryId);
+  public boolean queryCompleted(final long queryId) throws CatalogException {
+    return Status.finished(getQueryStatus(queryId).status);
   }
 
   /**

